bench-tps/src/main.rs

```diff
@@ -132,11 +133,12 @@ fn create_client(
     num_nodes: usize,
     target_node: Option<Pubkey>,
     connection_cache: ConnectionCache,
+    commitment_config: CommitmentConfig,
 ) -> Arc<dyn BenchTpsClient + Send + Sync> {
     match external_client_type {
         ExternalClientType::RpcClient => Arc::new(RpcClient::new_with_commitment(
             json_rpc_url.to_string(),
-            CommitmentConfig::confirmed(),
+            commitment_config,
```

Comment:
It isn't shown in the diff, but it would be nice to control the commitment level for `ThinClient` too. I think `ThinClient::new()` will default to `Finalized` but if you use `ThinClient::new_from_client()`, you can control the commitment in the underlying `RpcClient`

Reply:
Ahh, my suggestion came from seeing this:
https://github.com/solana-labs/solana/blob/3284d34e041f737ddb0b946d0695d9c6ada0fc71/thin-client/src/thin_client.rs#L152-L163

But it doesn't look like the wrapper in `client/src/thin_client.rs` exposes this constructor. So, I think you're right that we should consider exposing the other constructor in `client/src/thin_client.rs`. If we go that route, I think we should potentially do that in a separate PR, and then rebase this PR on top of that one when it lands

Reply:
ya looks like `ThinClient::new` calls `Self::new_from_client`.  Good idea. Ok I can wait and do that in a separate PR

Reply:
ok I'll make a new PR for this exposing new_from_client

Reply:
See PR: [#35294](https://github.com/solana-labs/solana/pull/35294) for `ThinClient` `CommitmentConfig` update